src/components/utils.ts

```diff
@@ -0,0 +1,33 @@
+class ManagedProps<T> {
+  private usedProps: (keyof T)[] = [];
```

Comment:
This should be a set, accessing the same property multiple times would push it multiple times into the array.

Reply:
Since we decided to go with the spread operator approach, is it ok if i remove this file, and resolve all your suggestion?

Reply:
Yes